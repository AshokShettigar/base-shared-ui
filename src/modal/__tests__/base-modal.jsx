import { shallow } from 'enzyme';
import BaseModal from 'platform-common-ui/modal/base-modal';

describe('modal/base-modal', function() {
    it('should render the component', function() {
        const wrapper = shallow(
            <BaseModal
                id="modal1"
                title="Modal Title"
            >
                Some content
            </BaseModal>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the component with onClose function', function() {
        const onClosePopup = jest.fn();
        const wrapper = shallow(
            <BaseModal
                id="modal1"
                title="Modal Title"
                onClosePopup={onClosePopup}
            >
                Some content
            </BaseModal>
        );
        expect(wrapper).toMatchSnapshot();
        wrapper.find('.cn-modal__navbar__close').simulate('click');
        expect(onClosePopup).toHaveBeenCalledTimes(1);
        expect(onClosePopup).toHaveBeenCalledWith('modal1');
    });

    it('should render the component with width and height', function() {
        const wrapper = shallow(
            <BaseModal
                height="200px"
                id="modal1"
                title="Modal Title"
                width="240px"
            >
                Some content
            </BaseModal>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
