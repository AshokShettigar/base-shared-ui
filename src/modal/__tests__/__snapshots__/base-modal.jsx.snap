// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal/base-modal should render the component 1`] = `
<div
  className="cn-modal__container"
  data-component="BaseModal"
  style={
    Object {
      "height": "450px",
      "width": "600px",
    }
  }
>
  <Navbar
    className="cn-modal__navbar"
    color="dark"
    full={true}
    tag="nav"
    toggleable={false}
  >
    <NavbarBrand
      tag="a"
    >
      Modal Title
    </NavbarBrand>
  </Navbar>
  Some content
</div>
`;

exports[`modal/base-modal should render the component with onClose function 1`] = `
<div
  className="cn-modal__container"
  data-component="BaseModal"
  style={
    Object {
      "height": "450px",
      "width": "600px",
    }
  }
>
  <Navbar
    className="cn-modal__navbar"
    color="dark"
    full={true}
    tag="nav"
    toggleable={false}
  >
    <NavbarBrand
      tag="a"
    >
      Modal Title
    </NavbarBrand>
    <Nav
      tag="ul"
    >
      <NavItem
        className="cn-modal__navbar__close"
        onClick={[Function]}
        tag="li"
      >
        <span
          className="a11y"
        >
          Close
        </span>
        <Icon
          className="cn-modal__navbar__icon"
          icon="close"
          size="extra-small"
        />
      </NavItem>
    </Nav>
  </Navbar>
  Some content
</div>
`;

exports[`modal/base-modal should render the component with width and height 1`] = `
<div
  className="cn-modal__container"
  data-component="BaseModal"
  style={
    Object {
      "height": "200px",
      "width": "240px",
    }
  }
>
  <Navbar
    className="cn-modal__navbar"
    color="dark"
    full={true}
    tag="nav"
    toggleable={false}
  >
    <NavbarBrand
      tag="a"
    >
      Modal Title
    </NavbarBrand>
  </Navbar>
  Some content
</div>
`;
